pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -418,9 +421,22 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 
 			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
 			if err != nil || pvc == nil {
-				// if the PVC is not found, log the error and count the PV towards the PV limit
-				klog.V(4).Infof("Unable to look up PVC info for %s/%s, assuming PVC matches predicate when counting limits: %v", namespace, pvcName, err)
-				filteredVolumes[pvID] = true
+				// If the PVC is invalid, We don't count the volume here because
+				// there's no guarantee that it belongs to the running predicate.
+				klog.V(4).Infof("Unable to look up PVC info for %s/%s, assuming PVC doesn't match predicate when counting limits: %v", namespace, pvcName, err)
+				continue
+			}
+
+			storageClass, err := c.scInfo.GetStorageClassInfo(*pvc.Spec.StorageClassName)
```

Comment:
Should we check storageclass after VolumeName?  We only need to look at StorageClass for delayed binding case right?

Reply:
Yeat, that's better. Updated